### ISSUE TITLE
mdtest: can no longer copy and paste error messages directly into mdtests

### ISSUE THREAD

**Issue by AlexWaygood** [astral-sh/ruff#14910]
I ran an mdtest locally, and it failed with this output:

```
  crates/red_knot_python_semantic/resources/mdtest/type_of/basic.md:142 unmatched assertion: error: [lint:invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"
```

Okay. I copied and pasted the error message straight from my terminal, adding ``# error: [lint:invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"`` above the offending line in the mdtest to make it pass. I ran the mdtest again, and now it said:

```
  crates/red_knot_python_semantic/resources/mdtest/type_of/basic.md:142 unmatched assertion: error: [lint:invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"
  crates/red_knot_python_semantic/resources/mdtest/type_of/basic.md:142 unexpected error: 11 [lint:invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"
```

Note that the "expected message" there is identical to the message that mdtest said it got, but the test still failed! What mdtest _actually_ wants me to do to the mdtest to make the test pass is this:

```diff
- # error: [lint:invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"
+ # error: [invalid-base] "Invalid class base with type `GenericAlias` (all bases must be a class, `Any`, `Unknown` or `Todo`)"
```

But there's nothing in the error message it's giving me that would indicate that. This is very confusing UX :-)

Cc. @MichaReiser -- I think this was caused by one of your recent changes to diagnostics infrastructure in red-knot

**Comment by AlexWaygood** [comment]
Thanks @MichaReiser!
